### PR TITLE
ML Case Study Inteview

### PR DESCRIPTION
**Pull request by agude**
Still need to:

- [x] Find a lead image.
- [x] Update the alt-text
- [x] Update the description.

### CODE REVIEW

File: _drafts/2021-01-18-data_science_interview_prep_case_study.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+  Tweet recipients and number of minutes per day with a Tweet sent are likely
+  good features.
+- **Bots have a controller.** Someone is benefiting from the spam, and they
+  have to control their bots. Features around logins might help here like
+  number of accounts seen from this IP address or device, similarity of login
+  time, etc.
+
+### Model Selection
+
+I try to start at the simplest model that will work when starting a new
```
Comment (Veldrina) [inline _drafts/2021-01-18-data_science_interview_prep_case_study.md:209]:
```suggestion
I try to start with the simplest model that will work when starting a new
```

File: _drafts/2021-01-18-data_science_interview_prep_case_study.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+features that you believe to be predictive it is generally overkill.
+
+One thing to consider when training is that the dataset is probably going to
+be wildly imbalanced. I would start by down-sampling (since we likely have
+millions of events), but would be ready to discuss other methods and trade
+offs.
+
+### Validation
+
+Validation is not too hard at this point. We focus on the offline metric we
```
Comment (Veldrina) [inline _drafts/2021-01-18-data_science_interview_prep_case_study.md:232]:
```suggestion
Validation is not too difficult at this point. We focus on the offline metric we
```

File: _drafts/2021-01-18-data_science_interview_prep_case_study.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+need to run the model in real-time when Tweets are posted. Instead we can run
+batches and can decide on the time between runs by looking at something like
+the characteristic time a spam bot takes to send out Tweets. We can add
+rate limiting to Tweet sending as well to slow the spam bots and give us more
+time to decide without impacting normal users.
+
+For deployment, I would start in **shadow mode**, which I [discussed in detail
+in another post][shadow_mode]. This would allow us to see how the model
+performs on real data without the risk of blocking good accounts. I would
+track it's performance using our online metrics: spam fraction and ops time
```
Comment (Veldrina) [inline _drafts/2021-01-18-data_science_interview_prep_case_study.md:253]:
```suggestion
track its performance using our online metrics: spam fraction and ops time
```

File: _drafts/2021-01-18-data_science_interview_prep_case_study.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+saved. I can compute these metrics both assuming the model does and does not
+block the flagged accounts and then compare them. If the comparison is
```
Comment (Veldrina) [inline _drafts/2021-01-18-data_science_interview_prep_case_study.md:255]:
```suggestion
saved. I would compute these metrics twice, once using the assumption that the model blocks flagged accounts, and once assuming that it does not block flagged accounts, and then compare the two outcomes. If the comparison is
```

File: _drafts/2021-01-18-data_science_interview_prep_case_study.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+through a practice case study and how I would work through it.
+
+[should_i_go]: {% post_url 2018-08-21-should_i_go_to_insight %}
+[interviewing]: {% post_url 2020-09-21-interviewing_for_data_science_positions_in_2020 %}
+[last_post]: {% post_url 2020-12-14-data_science_interview_prep_data_manipulation %}
+
+## My Approach
+
+Case study interviews are just conversations. This can make them tougher than
+they need to be because they lack the obvious structure of a coding interview
```
[the remark this answers is not in the document]
Comment (agude) [reply _drafts/2021-01-18-data_science_interview_prep_case_study.md:40]: The latter.

**Pull request merged**